feat(ipfs): documentation about ipfs project (#1584)
scaleway/docs-content: storage/ipfs/api-cli/ipfs-operations.mdx

---
meta:

>   title: Understanding IPFS Storage ipfs operations

Reviewer: C'est normal le "ipfs operations" ici ? same pour les lignes du bas

Author: Maybe juste  "Understanding IPFS Storage operations" or without ipfs operations

  description: This page shows IPFS Storage API ipfs operations.
content:
  h1: Understanding IPFS Storage ipfs operations
  paragraph: This page shows IPFS Storage API ipfs operations.
tags: ipfs storage ipfs-storage volume
dates:
  validation: 2023-04-23
  posted: 2023-04-23
categories:
  - storage
  - ipfs-storage
---

> Scaleway IPFS allows you to store content on the IPFS network via CID or URL. You can [create](/storage/ipfs/api-cli/volumes-operations) and manage your volume  via the [Scaleway Command Line Interface](https://github.com/scaleway/scaleway-cli).

Reviewer: ```suggestion
Scaleway IPFS allows you to store content on the IPFS network via CID (Content-ID) or URL. You can [create](/storage/ipfs/api-cli/volumes-operations) and manage your volume via the [Scaleway Command Line Interface](https://github.com/scaleway/scaleway-cli).
```

## Scaleway Command Line Interface Overview

[The Scaleway Command Line Interface (CLI)](https://github.com/scaleway/scaleway-cli) The Scaleway command line allows you to create and interact with your pins. You will be able to create volumes, manage them, add/delete content within them

<Macro id="iam-requirements" />

<Message type="requirement">
    - You have an account and are logged into the [Scaleway Console](https://console.scaleway.com/)
    - You have generated your [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
    - You have downloaded and installed the [Scaleway CLI v2](https://github.com/scaleway/scaleway-cli#scaleway-cli-v2)
</Message>

> ## AddCID

Reviewer: ```suggestion
## Adding CID
```

> This operation allows you to add content by CID in volume.

Reviewer: ```suggestion
This operation allows you to add content by CID in a volume.
```

**Sample request**
```sh no-copy
POST /ipfs/v1alpha1/regions/fr-par/pins/create-by-cid HTTP/1.1
Host: agw.stg.fr-par-2.internal.scaleway.com
User-Agent: scaleway-sdk-go/v1.0.0-beta.7+dev (go1.20.3; linux; amd64) scaleway-cli/2.14.0-1
Content-Length: 169
Content-Type: application/json
X-Auth-Token: 6a55c9c4-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

**CLI command**
```
scw ipfs ipfs add-cid volume-id=11111111-1111-1111-1111-111111111111 cid=Qmdi7ERksspfxWXfU8ATRUt7iCjZJbEbDrUoMDtjnbdTwo name=newcid
```

**Sample response**
```
PinID           22222222-2222-2222-2222-222222222222
Status          queued
CreatedAt       now
Cid.Cid         Qmdi7ERksspfxWXfU8ATRUt7iCjZJbEbDrUoMDtjnbdTwo
Cid.Name        newcid
Cid.Meta.AppID  00000000-0000-0000-0000-000000000000
```

> ## AddURL

Reviewer: ```suggestion
## Adding URL
```

> This operation allows you to add content by CID in volume.

Reviewer: ```suggestion
This operation allows you to add content by URL in a volume.
```

**Sample request**
```sh no-copy
POST /ipfs/v1alpha1/regions/fr-par/pins/create-by-url HTTP/1.1
Host: agw.stg.fr-par-2.internal.scaleway.com
User-Agent: scaleway-sdk-go/v1.0.0-beta.7+dev (go1.20.3; linux; amd64) scaleway-cli/2.14.0-1
Content-Length: 121
Content-Type: application/json
X-Auth-Token: 6a55c9c4-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

**CLI command**
```
scw ipfs ipfs add-url volume-id=11111111-1111-1111-1111-111111111111 url=http://labs.scaleway.com/ name=newcid
```

**Sample response**
```
PinID           22222222-2222-2222-2222-222222222222
Status          queued
CreatedAt       now
Cid.Cid         -
Cid.Name        newcid
Cid.Meta.AppID  00000000-0000-0000-0000-000000000000
```

<Message type="note">
  If the response is ingest you can retrieve the CID of your url.
</Message>

> ## GetPinID

Reviewer: ```suggestion
## Getting Pin ID
```

This operation allows you to get information about pin id.

<Message type="note">
  You can retrieve CID associed about pin-id or discover cid behind your url.
</Message>

**Sample request**
```sh no-copy
GET /ipfs/v1alpha1/regions/fr-par/pins/22222222-2222-2222-2222-222222222222?volume_id=11111111-1111-1111-1111-111111111111 HTTP/1.1
Host: agw.stg.fr-par-2.internal.scaleway.com
User-Agent: scaleway-sdk-go/v1.0.0-beta.7+dev (go1.20.3; linux; amd64) scaleway-cli/2.14.0-1
X-Auth-Token: 6a55c9c4-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

**CLI command**
```
scw ipfs ipfs get-pin-id pin-id=22222222-2222-2222-2222-222222222222 volume-id=11111111-1111-1111-1111-111111111111
```

**Sample response**
```
Status          pinned
CreatedAt       5 minutes ago
Cid.Cid         QmaGNR5PedL8kiTUMtsPeFFodL9HnYBMuUqAEQrv6CQH4F
Cid.Name        value:"newcid"
Cid.Meta.AppID  00000000-0000-0000-0000-000000000000
```

<Message type="note">
  If the response is ingest you can retrieve the CID of your url.
</Message>

<Message type="important">

> If status is failed, you can verify information about CID or URL and retry.

Reviewer: ```suggestion
If the status is failed, you can verify information about the CID or the URL and retry.
```
